src/lsm/binary_search.zig

```diff
@@ -30,7 +37,7 @@ pub fn binary_search_values_raw(
 ) u32 {
     if (values.len == 0) return 0;
 
-    if (config.verify) {
+    if (constants.verify) {
```

Comment:
Hm, not sure that using `constants.verify` is right here, because it changes big-O complexity from `log(N)` to `N`, so this'll probably slow down simulator and such too much. 

It makes sense that binary search has it's own toggle for full verification.

Reply:
All the other instancies of verify I think are fine though, as they preserve logarithmic behavior.

Reply:
yeah, let's just remove it. This assert checks the caller, but if the caller are tests, then it doesn't really matter.